bip-0360.mediawiki

```diff
@@ -11,696 +11,397 @@
-upgraded by 2030, with browsers and operating systems fully upgraded by 2033. According to NIST IR 8547, Elliptic Curve
-Cryptography is planned to be disallowed within the US federal government after 2035. An exception is made for hybrid
-cryptography, which is the use of ECC and post-quantum algorithms together.
+Since P2TSH outputs have no key path spend, they omit the Taproot internal key. Instead, a P2TSH output includes the 32-byte root of the tapleaf Merkle tree as defined in [[bip-0341.mediawiki|BIP 341]] hashed with the tag "TapBranch" as shown below.
 
-Although the main threat posed by CRQCs is to the signatures used in Bitcoin, a smaller threat is to Bitcoin's hash
-algorithms. In particular, while a CRQC could use [https://en.wikipedia.org/wiki/Grover's_algorithm Grover's algorithm]
-to gain a quadratic speedup on brute-force attacks on the hash functions used in Bitcoin, a significantly more powerful
-CRQC is needed for these attacks to meaningfully impact Bitcoin. For instance, a preimage attack on
-HASH160 <ref name="hash160">Used by P2PKH, P2SH, and P2WPKH addresses, though not P2WSH because it uses 256-bit hashes.</ref>
-using Grover's algorithm would require at least 10^24 quantum operations. As for Grover's application to mining, see
-[https://quantumcomputing.stackexchange.com/a/12847 Sam Jaques' post on this].
+[[File:bip-0360/media/merkletree.png|thumb|Construction of P2TSH Taptree root, scriptPubkey, and Witness]]
 
-=== Design ===
-
-This BIP proposes a new output type called P2QRH (Pay to Quantum Resistant Hash). This output type is designed to 
-support post-quantum signature algorithms but those algorithms will be specified in future BIPs.
-
-P2QRH (Pay to Quantum Resistant Hash) is a new output type that commits to the root of a tapleaf Merkle tree. It is functionally
-the same as a P2TR (Pay to Taproot) output with the quantum vulnerable key path spend removed. Since P2QRH has no key path spend, P2QRH omits the
-Taproot internal key, as it is not needed. Instead, a P2QRH output is just the 32-byte root of the tapleaf Merkle tree as defined
-in [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki BIP 341] and hashed with the tag "QuantumRoot" as shown below.
-
-[[File:bip-0360/merkletree.png|center|550px|thumb|]]
-
-To construct a P2QRH output we follow the same process as [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki BIP 341]
-to compute the tapscript Merkle root. However, instead of the root of the Merkle tree being hashed together with the internal
-key in P2QRH the root is hashed by itself using the tag "QuantumRoot".
+To construct a P2TSH output, we follow the process outlined in [[bip-0341.mediawiki|BIP 341]] to compute the final tapbranch hash, which is the merkle root of the tapleaves; however, instead of tweaking the internal key with the root of the Merkle tree (as is the case with P2TR outputs), P2TSH outputs commit only to final tapbranch hash, which is tagged, "TapBranch".
```

Comment:
suggest taking a second look at this. What happens in Taproot is:

taproot = internal_pubkey + H("TapTweak" || internal_pubkey || taptree_root) * G

There's some language in https://github.com/cryptoquick/bips/pull/31 that describes this.

Reply:
I see what you're saying now. We already link to BIP 341, so in my opinion, those details don't need to be repeated here, especially because that's not the point we're trying to make.

Reply:
Ok, sounds good. As an alternative you could just say 'instead of tweaking the internal pubkey as is the case with P2TR' and leave it at that. My point here is that Taproot doesn't tweak the internal pubkey with the root of any Merkle tree as is stated.